istioctl/cmd/sidecar.go

```diff
@@ -0,0 +1,361 @@
+// Copyright Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"archive/tar"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/spf13/cobra"
+
+	networkingv1alpha3 "istio.io/api/networking/v1alpha3"
+	clientv1alpha3 "istio.io/client-go/pkg/apis/networking/v1alpha3"
+	"istio.io/istio/pkg/config/mesh"
+	"istio.io/istio/pkg/config/schema/collections"
+	"istio.io/istio/pkg/kube"
+	"istio.io/istio/pkg/kube/inject"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+var (
+	name           string
+	serviceAccount string
+	filename       string
+	outputName     string
+	clusterID      string
+	ports          []string
+)
+
+const (
+	tempPerms = os.FileMode(0644)
+)
+
+func sidecarCommands() *cobra.Command {
+	sidecarCmd := &cobra.Command{
+		Use:   "sidecar",
+		Short: "Commands to assist in managing sidecar configuration",
+	}
+	sidecarCmd.AddCommand(createGroupCommand())
+	sidecarCmd.AddCommand(generateConfigCommand())
+	return sidecarCmd
+}
+
+func createGroupCommand() *cobra.Command {
+	createGroupCmd := &cobra.Command{
+		Use:   "create-group",
+		Short: "Creates a WorkloadGroup YAML artifact representing workload instances",
+		Long: `Creates a WorkloadGroup YAML artifact representing workload instances for passing to the Kubernetes API server.
```

Comment:
better Long:
`Creates a WorkloadGroup API YAML artifact to send to the Kubernetes API server.`